optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -380,11 +380,11 @@ def pre_attn_forward(
                 past_key_value = (self.k_cache.get_shape(), self.v_cache.get_shape())
             else:
                 if past_key_value is None:
-                    past_key = torch.zeros(key_states.shape, dtype=self.k_proj.weight.dtype, device=key_states.device)
                     past_value = torch.zeros(
                         key_states.shape, dtype=self.k_proj.weight.dtype, device=key_states.device
                     )
```

Comment:
only past_value is defined, past_key is not.

it needs to be dfined like here: https://github.com/HabanaAI/optimum-habana-fork/blob/habana-main/optimum/habana/transformers/models/llama/modeling_llama.py#L304